wasm/jit/engine.go

```diff
@@ -113,121 +119,45 @@ func (e *engine) Call(f *wasm.FunctionInstance, params ...uint64) (results []uin
 				e.callFrameStack = nil
 				e.stackBasePointer = 0
 				e.stackPointer = 0
+				e.callFrameNum = 0
 			}
 		}
 	}()
 
 	for _, param := range params {
 		e.push(param)
 	}
-	// Note that there's no conflict between e.hostFunctionIndex and e.compiledWasmFunctionIndex,
-	// meaning that each *wasm.FunctionInstance is assigned to either host function index or wasm function one.
-	if index, ok := e.compiledHostFunctionIndex[f]; ok {
-		e.compiledHostFunctions[index].f(&wasm.HostFunctionCallContext{Memory: f.ModuleInstance.Memory})
-	} else if index, ok := e.compiledWasmFunctionIndex[f]; ok {
-		f := e.compiledWasmFunctions[index]
-		e.exec(f)
-	} else {
+
+	compiled, ok := e.compiledFunctions[f.Address]
+	if !ok {
 		err = fmt.Errorf("function not compiled")
 		return
 	}
+
+	if compiled.isHostFunction() {
+		e.execHostFunction(compiled, &wasm.HostFunctionCallContext{Memory: f.ModuleInstance.Memory})
+	} else {
+		e.execWasmFunction(compiled)
```

Comment:
maybe we can only clarify when it isn't a wasm function. Ex execHostFunction is different so includes the name host, otherwise we assume wasm.

```suggestion
		e.execFunction(compiled)
```